### PR TITLE
RavenDB-15601 Additional option to collect debug-info

### PR DESCRIPTION
**Pull request by YonatanResh**
### Issue link

https://issues.hibernatingrhinos.com/issue/RavenDB-15601/Additional-option-to-collect-debug-info

### Additional description

in the UI we need to add the option to choose if you want to get the server wide, log file, and all the databases or a subset of the databases

### Type of change

- New feature

### How risky is the change?

- Low 

### Backward compatibility

- Non breaking change

### Testing by Contributor

- It has been verified by manual testing

### UI work

- It requires further work in the Studio

### CODE REVIEW

**Review by grisha-kotler — APPROVED** [review]

File: src/Raven.Server/Documents/Handlers/Debugging/ServerWideDebugInfoPackageHandler.cs [hunk 1]
```diff
@@ -227,9 +255,17 @@ await using (var sw = new StreamWriter(entryStream))
...
+        private async Task<Stream> GetDebugInfoFromNodeAsync(JsonOperationContext context, RequestExecutor requestExecutor, long operationId, OperationCancelToken token, int timeoutInSec, string type, StringValues databases)
         {
-            var rawStreamCommand = new GetRawStreamResultCommand($"/admin/debug/info-package?operationId={operationId}");
+            
+            var url = $"/admin/debug/info-package?operationId={operationId}";
+            if (type != null)
+                url += $"&type={type}";
+
+            if (string.IsNullOrEmpty(databases) == false)
+                url += $"&databases=${string.Join("&databases=", databases)}";
```
Comment (ppekrol) [inline src/Raven.Server/Documents/Handlers/Debugging/ServerWideDebugInfoPackageHandler.cs]: name needs to be encoded

File: src/Raven.Server/Documents/Handlers/Debugging/ServerWideDebugInfoPackageHandler.cs [hunk 2]
```diff
@@ -449,5 +497,15 @@ internal class NodeDebugInfoRequestHeader
 
             public List<string> DatabaseNames { get; set; }
         }
+
+        [Flags]
+        public enum DebugInfoContents
```
Comment (ppekrol) [inline src/Raven.Server/Documents/Handlers/Debugging/ServerWideDebugInfoPackageHandler.cs]: DebugInfoPackageContent

File: src/Raven.Server/Documents/Handlers/Debugging/ServerWideDebugInfoPackageHandler.cs [hunk 3]
```diff
@@ -292,6 +328,18 @@ private async Task WriteForAllLocalDatabases(ZipArchive archive, JsonOperationCo
                         await WriteDatabaseInfo(archive, jsonOperationContext, localEndpointClient, databaseName, token);
                     }
                 }
+
+                HashSet<string> GetDatabases()
+                {
+                    var allDatabases = ServerStore.Cluster.GetDatabaseNames(context);
+
+                    if (string.IsNullOrEmpty(databases) == false && databases.Any())
```
Comment (ppekrol) [inline src/Raven.Server/Documents/Handlers/Debugging/ServerWideDebugInfoPackageHandler.cs]: this should be `databases.Count > 0`, it is a struct

File: src/Raven.Server/Documents/Handlers/Debugging/ServerWideDebugInfoPackageHandler.cs [hunk 4]
```diff
@@ -167,6 +170,8 @@ public async Task GetInfoPackage()
             HttpContext.Response.Headers["Content-Disposition"] = contentDisposition;
             HttpContext.Response.Headers["Content-Type"] = "application/zip";
 
+            var debugInfoType = GetDebugInfoType();
+            var databases = GetStringValuesQueryString("databases", required: false);
```
Comment (ppekrol) [inline src/Raven.Server/Documents/Handlers/Debugging/ServerWideDebugInfoPackageHandler.cs]: by convention, this should be `database` not `databases`

you are passing via url "database=X&database=Y" - passing 'databases' does not look natural here

File: src/Raven.Server/Documents/Handlers/Debugging/ServerWideDebugInfoPackageHandler.cs [hunk 5]
```diff
@@ -108,8 +109,10 @@ public async Task GetClusterWideInfoPackage()
 
             var timeoutInSecPerNode = GetIntValueQueryString("timeoutInSecPerNode", false) ?? 60;
             var clusterOperationToken = CreateOperationToken();
+            var type = GetStringQueryString("type", false);
```
Comment (ppekrol) [inline src/Raven.Server/Documents/Handlers/Debugging/ServerWideDebugInfoPackageHandler.cs]: use named paramter

Comment (YonatanResh) [reply src/Raven.Server/Documents/Handlers/Debugging/ServerWideDebugInfoPackageHandler.cs]: done

File: src/Raven.Server/Documents/Handlers/Debugging/ServerWideDebugInfoPackageHandler.cs (hunk 4 above)
Comment (YonatanResh) [reply src/Raven.Server/Documents/Handlers/Debugging/ServerWideDebugInfoPackageHandler.cs]: done

File: src/Raven.Server/Documents/Handlers/Debugging/ServerWideDebugInfoPackageHandler.cs (hunk 3 above)
Comment (YonatanResh) [reply src/Raven.Server/Documents/Handlers/Debugging/ServerWideDebugInfoPackageHandler.cs]: done

File: src/Raven.Server/Documents/Handlers/Debugging/ServerWideDebugInfoPackageHandler.cs (hunk 2 above)
Comment (YonatanResh) [reply src/Raven.Server/Documents/Handlers/Debugging/ServerWideDebugInfoPackageHandler.cs]: done

File: src/Raven.Server/Documents/Handlers/Debugging/ServerWideDebugInfoPackageHandler.cs (hunk 1 above)
Comment (YonatanResh) [reply src/Raven.Server/Documents/Handlers/Debugging/ServerWideDebugInfoPackageHandler.cs]: done

File: src/Raven.Server/Documents/Handlers/Debugging/ServerWideDebugInfoPackageHandler.cs [hunk 6]
```diff
@@ -227,9 +256,16 @@ await using (var sw = new StreamWriter(entryStream))
...
-        private async Task<Stream> GetDebugInfoFromNodeAsync(JsonOperationContext context, RequestExecutor requestExecutor, long operationId, OperationCancelToken token, int timeoutInSec)
+        private async Task<Stream> GetDebugInfoFromNodeAsync(JsonOperationContext context, RequestExecutor requestExecutor, long operationId, OperationCancelToken token, int timeoutInSec, string type, StringValues databases)
         {
-            var rawStreamCommand = new GetRawStreamResultCommand($"/admin/debug/info-package?operationId={operationId}");
+            var url = $"/admin/debug/info-package?operationId={operationId}";
+            if (type != null)
+                url += $"&type={type}";
+
+            if (databases.Count > 0)
+                url += $"&database={WebUtility.UrlEncode(string.Join("&database=", databases))}";
```
Comment (ppekrol) [inline src/Raven.Server/Documents/Handlers/Debugging/ServerWideDebugInfoPackageHandler.cs]:
```
Uri.EscapeDataString(_indexName)
```

Comment (YonatanResh) [reply src/Raven.Server/Documents/Handlers/Debugging/ServerWideDebugInfoPackageHandler.cs]: done

**Pull request merged**